sys/include/net/nanocoap_sock.h

```diff
@@ -154,18 +160,99 @@ extern "C" {
 int nanocoap_server(sock_udp_ep_t *local, uint8_t *buf, size_t bufsize);
 
 /**
- * @brief   Simple synchronous CoAP (confirmable) get
+ * @brief   Create a CoAP client socket
  *
+ * @param[out]  sock    CoAP UDP socket
+ * @param[in]   local   Local UDP endpoint, may be NULL
  * @param[in]   remote  remote UDP endpoint
+ *
+ * @returns     0 on success
+ * @returns     <0 on error
+ */
+int nanocoap_sock_connect(nanocoap_sock_t *sock, sock_udp_ep_t *local,
+                          sock_udp_ep_t *remote);
+
+/**
+ * @brief   Close a CoAP client socket
+ *
+ * @param[in]  sock     CoAP UDP socket
+ */
+static inline void nanocoap_sock_close(nanocoap_sock_t *sock)
+{
+    sock_udp_close(sock);
+}
+
+/**
+ * @brief   Simple synchronous CoAP (confirmable) get
+ *
+ * @param[in]   sock    socket to use for the request
  * @param[in]   path    remote path
  * @param[out]  buf     buffer to write response to
  * @param[in]   len     length of @p buffer
  *
  * @returns     length of response payload on success
  * @returns     <0 on error
  */
-ssize_t nanocoap_get(sock_udp_ep_t *remote, const char *path, uint8_t *buf,
-                     size_t len);
+ssize_t nanocoap_sock_get(nanocoap_sock_t *sock, const char *path, void *buf,
```

Comment:
the old `nanocoap_get()` is dropped?

Reply:
It's still [there](https://github.com/RIOT-OS/RIOT/pull/17474/files#diff-06979ffde8298a874f48faabf33bebe43f95bc5b245bc8130ea07b3e597d2409R284)